pkg/state/state_hostdevice_network.go

```diff
@@ -55,7 +55,7 @@ func NewStateHostDeviceNetwork(k8sAPIClient client.Client, manifestDir string) (
 			description: stateHostDeviceNetworkDescription,
 			client:      k8sAPIClient,
 			renderer:    renderer,
-		}}, nil
+		}}, nil, nil
```

Comment:
no need to return state as MainfestRenderer instead of nil? (here and in other modified states in this PR)

in hindsight perhaps we should have embedded MainifestRenderer in State interface and just return one value.
but that not related to this PR.

Reply:
The problem is that if we also want to return the renderer here we have to adapt the `getManifestObjects` and make it also public. We can do this for sure, but it is not needed and we can simply return `nil` here.
I don't have a real opinion about that, but I can also change it to return the renderer here as well. WDYT?

Reply:
Ah i see, i though all states implement MainfestRenderer, i see this is not the case.

ack, lets keep it as is.